### PR TITLE
limit the H3 resolution we test until lucene bug is fixed

### LINKED ISSUE #91915: [CI] ParentChildNavigationTests testNoChildrenIntersecting failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/vjzfcjaubmwxc/tests/:libs:elasticsearch-h3:test/org.elasticsearch.h3.ParentChildNavigationTests/testNoChildrenIntersecting

**Reproduction line:**
```
./gradlew ':libs:elasticsearch-h3:test' --tests "org.elasticsearch.h3.ParentChildNavigationTests.testNoChildrenIntersecting" -Dtests.seed=1655A5B15780FE8D -Dtests.locale=ko-KR -Dtests.timezone=Europe/Saratov -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.h3.ParentChildNavigationTests&tests.test=testNoChildrenIntersecting

**Failure excerpt:**
```
java.lang.AssertionError: expected:<2> but was:<3>

  at __randomizedtesting.SeedInfo.seed([1655A5B15780FE8D:CE3E9CBB8EC2EC9D]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.failNotEquals(Assert.java:834)
  at org.junit.Assert.assertEquals(Assert.java:645)
  at org.junit.Assert.assertEquals(Assert.java:631)
  at org.elasticsearch.h3.ParentChildNavigationTests.assertIntersectingChildren(ParentChildNavigationTests.java:96)
  at org.elasticsearch.h3.ParentChildNavigationTests.testNoChildrenIntersecting(ParentChildNavigationTests.java:81)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by iverase**
This test is failing due to lucene bug https://github.com/apache/lucene/issues/11883. The issue has been fixed so until a new lucene snapshot is used, let's limit the resolutions we test in ParentChildNavigationTests#testNoChildrenIntersecting.

fixes https://github.com/elastic/elasticsearch/issues/91915

### CODE REVIEW

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java [hunk 1]
```diff
@@ -132,13 +133,23 @@ private void assertHexRing(int res, String h3Address, String[] children) {
...
+        for (int i = 1; i <= 10; i++) {
             h3Addresses = H3.h3ToChildren(h3Address);
             assertIntersectingChildren(h3Address, h3Addresses);
             h3Address = RandomPicks.randomFrom(random(), h3Addresses);
         }
     }
 
+    public void testIssue91915() {
+        GeoPolygon polygon1 = getGeoPolygon("8cc373cb54069ff");
+        GeoPolygon polygon2 = getGeoPolygon("8cc373cb54069ff");
```
Comment (craigtaverner) [inline libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:146]: The h3 code is identical, so these two polygons are identical (same H3 cell). Perhaps you meant one of them to be different?

**Review by craigtaverner — CHANGES_REQUESTED** [review]
I think there is a mistake in the one test.

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java (hunk 1 above)
Comment (iverase) [reply libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java]: That was a copy/paste error, I updated it to add the right cell. True that the original issue cannot be reproduced anymore, no idea why but here is another error if you want to have a look:

```
./gradlew ':libs:elasticsearch-h3:test' --tests "org.elasticsearch.h3.ParentChildNavigationTests.testNoChildrenIntersecting" -Dtests.seed=BD263AFF9F3817AA -Dtests.locale=ca-ES -Dtests.timezone=America/Merida -Druntime.java=1
```

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by craigtaverner — APPROVED** [review]
LGTM

**Pull request merged**